File: Python/Algorithms/690.py
# Definition for Employee.
class Employee:
    def __init__(self, id: int, importance: int, subordinates):
        self.id = id
        self.importance = importance
        self.subordinates = subordinates


class Solution:
    def getImportance(self, employees, id):
        d = {e.id: [e.importance, e.subordinates] for e in employees
                if id not in e.subordinates
        }
        # get importance and subordinates
        importance, subords = d[id]

        while subords:
            subord = subords.pop()
            si, ss = d[subord]

            importance = importance + si
            if ss:
                subords.extend(ss)
        return importance

File: Javascript/Algorithms/690.js
/**
 * Definition for Employee.
 * function Employee(id, importance, subordinates) {
 *     this.id = id;
 *     this.importance = importance;
 *     this.subordinates = subordinates;
 * }
 */

let GetImportance = (employees, id) => {
    const empls = new Map;
    for (let e of employees){
        empls.set(e.id, [e.importance, e.subordinates]);
    }
    
    let [importance, subords] = empls.get(id);
    while (subords.length){
        let subord_id = subords.pop();
        let [s_imp, s_subords] = empls.get(subord_id);
        
        importance += s_imp;
        if (s_subords.length){
            subords.push(...s_subords);
        }
    }
    return importance;
};
